polar-core/tests/integration_tests.rs

```diff
@@ -1568,8 +1568,18 @@ fn test_rest_vars() -> TestResult {
     qeval(&mut p, "append([1,2,3], [], [1,2,3])");
     qeval(&mut p, "not append([1,2,3], [4], [1,2,3])");
 
+    qeval(
+        &mut p,
+        "a = [1, *b] and b = [2, *c] and c=[3] and 1 in a and 2 in a and 3 in a",
+    );
+
     let a = &var(&mut p, "[*_] in [*a] and [*b] in [*_] and b = 1", "a")[0];
-    assert!(matches!(a, Value::List(b) if matches!(b[0].value(), Value::RestVariable(_))));
+    assert!(matches!(a,
```

Comment:
So I updated this test to expect a new expression, and the expression seems to be correct, but actually I'm a little confused about what we're trying to test here. The constraint described by the expression is basically "a is a list containing a list".